### PR TITLE
Add structured logging and Prometheus events

### LINKED ISSUE #1382: Validators should be able to operate the attestation service with proper logging/monitoring
### Expected Behavior

Please describe the behavior you are expecting

### Current Behavior

What is the current behavior?

### PR DESCRIPTION
**Pull request by nambrot**
### Description

This PR adds structured logging and Prometheus events to the attestation service.

### Tested

- Locally against `namoffchainreveal`

### Related issues

- Fixes #1382

### CODE REVIEW

File: packages/attestation-service/src/index.ts [hunk 1]
```diff
@@ -1,38 +1,46 @@
 import * as dotenv from 'dotenv'
+
+// We need to load the config before some of the imports for e.g. the logger
+if (process.env.CONFIG) {
```
Comment (mcortesi) [inline packages/attestation-service/src/index.ts]: this doesn't sound right... imports by ES6 definition must be the first thing on a file, you can't add imports after code

File: packages/attestation-service/src/index.ts [hunk 2]
```diff
@@ -1,38 +1,46 @@
...
+  dotenv.config({ path: process.env.CONFIG })
+}
+
 import express from 'express'
 import RateLimiter from 'express-rate-limit'
+import requestIdMiddleware from 'express-request-id'
+import * as PromClient from 'prom-client'
 import { initializeDB, initializeKit } from './db'
-import { createValidatedHandler } from './request'
+import { logger } from './logger'
```
Comment (mcortesi) [inline packages/attestation-service/src/index.ts]: maybe rename to "rootLogger"

File: packages/attestation-service/src/logger.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    : { stream: process.stdout, level: logLevel }
+
+export const logger = createLogger({
+  name: 'attestation-service',
+  serializers: stdSerializers,
+  streams: [stream],
+})
+
+// Somehow it wouldn't let me export logger without export Logger
+export { Logger }
```
Comment (mcortesi) [inline packages/attestation-service/src/logger.ts]: try setting the type for const logger explicitely

`const logger: Logger`

File: packages/attestation-service/src/request.ts [hunk 4]
```diff
@@ -1,29 +1,33 @@
...
 import express from 'express'
 import { isLeft } from 'fp-ts/lib/Either'
 import * as t from 'io-ts'
+import { logger } from './logger'
 
 export enum ErrorMessages {
   UNKNOWN_ERROR = 'Something went wrong',
 }
 
 export function catchAsyncErrorHandler<T>(
```
Comment (mcortesi) [inline packages/attestation-service/src/request.ts]: i think `asyncHandler()` is good enough name :P

File: packages/attestation-service/src/request.ts [hunk 5]
```diff
@@ -64,3 +68,22 @@ function serializeErrors(errors: t.Errors) {
 export function respondWithError(res: express.Response, statusCode: number, error: string) {
   res.status(statusCode).json({ success: false, error })
 }
+
+export type Response = Omit<express.Response, 'locals'> & {
+  locals: { logger: Logger } & any
```
Comment (mcortesi) [inline packages/attestation-service/src/request.ts]: why the `& any`?

File: packages/attestation-service/src/request.ts [hunk 6]
```diff
@@ -1,29 +1,33 @@
...
 ) {
-  return async (req: express.Request, res: express.Response) => {
-    try {
-      return handler(req, res)
-    } catch (error) {
-      console.error(error)
-      respondWithError(res, 500, ErrorMessages.UNKNOWN_ERROR)
-    }
+  return (req: express.Request, res: Response) => {
+    handler(req, res)
```
Comment (mcortesi) [inline packages/attestation-service/src/request.ts]: word of caution.... here you don't catch an edge case, suppose the handler() do some things and then return a promise... for example:

```ts
function evilHandler() {
  if (true) {
    throw new Error()
  }
  return callSomethingThatRetunsAPromise()
}
```

In this case... there's a throw before, you are not actually catching the error.

**Review by mcortesi — COMMENTED** [review]
left some comments

File: packages/attestation-service/src/request.ts (hunk 5 above)
Comment (nambrot) [reply packages/attestation-service/src/request.ts]: I wasn't entirely sure if the compiler would be capable to intersect both `locals`, but let me try

File: packages/attestation-service/package.json [hunk 7]
```diff
@@ -31,6 +31,8 @@
     "@celo/utils": "^0.1.0",
     "bignumber.js": "^7.2.0",
     "body-parser": "1.19.0",
+    "bunyan": "1.8.12",
+    "bunyan-gke-stackdriver": "0.1.2",
```
Comment (jmrossy) [inline packages/attestation-service/package.json:35]: Doesn't this couple the service tightly with GCloud? Validators may want to host this on AWS/Azure

File: packages/attestation-service/src/request.ts (hunk 5 above)
Comment (nambrot) [reply packages/attestation-service/src/request.ts]: Turns out 

```
export type Response = Omit<express.Response, 'locals'> & {
  locals: { logger: Logger } & Omit<any, 'logger'>
}
```

was the way to go

File: packages/attestation-service/src/request.ts (hunk 6 above)
Comment (nambrot) [reply packages/attestation-service/src/request.ts]: Great catch, definitely wasn't aware

File: packages/attestation-service/package.json (hunk 7 above)
Comment (nambrot) [reply packages/attestation-service/package.json:35]: `bunyan-gke-stackdriver` by @mcortesi is a plugin to `bunyan` to create an alternatively formatted output stream, it does not require any GCP specific functionality to build/run, and can be configured with `LOG_FORMAT=stackdriver`, so is opt-in

**Review by mcortesi — APPROVED** [review]

**Comment by nambrot** [comment]
Tests pass, but enabled SSH

**Pull request merged**